### PR TITLE
Adding basePath as a parameter that can be passed in through the POM 

### PR DESCRIPTION
**Pull request by aaronSig**

### CODE REVIEW

**Comment by loosebazooka** [comment]
@aaronSign, @tangiel I'm not familiar with alternative basepaths, what's the usecase here?

File: pom.xml [hunk 1]
```diff
@@ -4,7 +4,7 @@
 
   <groupId>com.google.cloud.tools</groupId>
   <artifactId>endpoints-framework-maven-plugin</artifactId>
-  <version>1.0.2-SNAPSHOT</version>
+  <version>1.0.3-SNAPSHOT</version>
```
Comment (loosebazooka) [inline pom.xml]: please revert this change to version.

**Comment by aaronSig** [comment]
For me it's an effort to create a more intuitive URL structure. I'm creating a REST API that's being consumed by 3rd parties.

The default basePath is `/_ah/api` which is added to all the urls:
`example.com/_ah/api/v1/users`

Then removing the base url I can get:
`example.com/v1/users`

Which is a bit easier on the eye.

File: pom.xml (hunk 1 above)
Comment (aaronSig) [reply pom.xml]: Sure. All done

**Comment by aaronSig** [comment]
I signed it!

**Comment by loosebazooka** [comment]
I'm going to run it by @tangiel, since he leads the core endpoints development work.

File: src/main/java/com/google/cloud/tools/maven/endpoints/framework/OpenApiDocsMojo.java [hunk 2]
```diff
@@ -18,21 +18,17 @@
 package com.google.cloud.tools.maven.endpoints.framework;
 
 import com.google.api.server.spi.tools.EndpointsTool;
-import com.google.api.server.spi.tools.GetDiscoveryDocAction;
 import com.google.api.server.spi.tools.GetOpenApiDocAction;
 import com.google.common.base.Joiner;
 import com.google.common.base.Strings;
+import org.apache.maven.plugin.MojoExecutionException;
```
Comment (loosebazooka) [inline src/main/java/com/google/cloud/tools/maven/endpoints/framework/OpenApiDocsMojo.java]: looks like I never added the formatter/checkstyle plugins to this repo :(.

can you run `mvn com.coveo:fmt-maven-plugin:format` to make sure it's google java format compliant?

Comment (aaronSig) [reply src/main/java/com/google/cloud/tools/maven/endpoints/framework/OpenApiDocsMojo.java]: The formatting modifies about 10 files. Would it be better done in another pull request so we don't lose the flavour of this commit and make review a bit easier? 

If not happy to add.

Comment (loosebazooka) [reply src/main/java/com/google/cloud/tools/maven/endpoints/framework/OpenApiDocsMojo.java]: Oh right, good point. Well I would like at least the new files to follow the format. Maybe just run it and then revert the files that we don't want included in the PR.

```
$ mvn com.coveo:fmt-maven-plugin:format
$ git checkout -- <files we want to exclude from this request>
```

or maybe just revert that "include" section.

**Comment by loosebazooka** [comment]
Thanks.

**Comment by tangiel** [comment]
Sounds good to me! Thanks for the contribution!

File: src/main/java/com/google/cloud/tools/maven/endpoints/framework/OpenApiDocsMojo.java [hunk 3]
```diff
@@ -33,30 +32,32 @@
 import org.apache.maven.plugins.annotations.Parameter;
 import org.apache.maven.plugins.annotations.ResolutionScope;
 import org.apache.maven.project.MavenProject;
+import org.apache.maven.project.MavenProject;
```
Comment (loosebazooka) [inline src/main/java/com/google/cloud/tools/maven/endpoints/framework/OpenApiDocsMojo.java]: This is a duplicate.

**Comment by loosebazooka** [comment]
minor code issue, I'll merge in after that. Thanks again for your contribution.

File: src/main/java/com/google/cloud/tools/maven/endpoints/framework/OpenApiDocsMojo.java (hunk 3 above)
Comment (aaronSig) [reply src/main/java/com/google/cloud/tools/maven/endpoints/framework/OpenApiDocsMojo.java]: yep, good spot!
